packages/ui-extensions/src/surfaces/admin/components/CustomerSegmentTemplate/CustomerSegmentTemplate.doc.ts

```diff
@@ -0,0 +1,42 @@
+import {ReferenceEntityTemplateSchema} from '@shopify/generate-docs';
+
+const data: ReferenceEntityTemplateSchema = {
+  name: 'CustomerSegmentTemplate',
+  description:
+    'CustomerSegmentTemplate is used to configure a template used in the customers section in the admin.',
+  requires:
+    'use of the [admin.customers.segmentation-templates.render](/docs/api/admin-extensions/api/extension-targets#extensiontargets-propertydetail-admincustomerssegmentationtemplatesrender) target.',
+  isVisualComponent: true,
+  thumbnail: 'customersegmenttemplate-thumbnail.png',
+  type: '',
+  definitions: [
+    {
+      title: 'CustomerSegmentTemplateProps',
+      description: '',
+      type: 'CustomerSegmentTemplateProps',
+    },
+  ],
+  category: 'Components',
+  subCategory: 'Other',
+  defaultExample: {
+    image: 'customersegmenttemplate-default',
+    codeblock: {
+      title: 'Simple CustomerSegmentTemplate implementation',
+      tabs: [
+        {
+          title: 'React',
+          code: './examples/CustomerSegmentTemplate.example.tsx',
+          language: 'tsx',
+        },
+        {
+          title: 'JS',
+          code: './examples/CustomerSegmentTemplate.example.ts',
```

Comment:
I am also wondering if we should remove the second template from this example. After talking with @vividviolet, we wanted to force a 1:1 relationship between an extension and a template. I feel like this is encouraging devs to do otherwise. WDYT?

This does not have to be addressed here but I think we might want to provide the same template as in the react example.

Reply:
Hmm, all the other components have examples in React and JS so I was following suit. I see what you mean. Do you think making the examples the same (ie they would render the same template) would be better?